javascript/ql/src/semmle/javascript/PackageExports.qll

```diff
@@ -68,10 +68,22 @@ DataFlow::Node getAValueExportedBy(PackageJSON packageJSON) {
 private DataFlow::Node getAnExportFromModule(Module mod) {
   result.analyze().getAValue() = mod.(NodeModule).getAModuleExportsValue()
   or
-  exists(ASTNode export | result.getEnclosingExpr() = export | mod.exports(_, export))
+  result = getAnExportedValue(mod, _)
+}
+
+/**
+ * Gets a value exported from `mod` under `name`.
+ */
+DataFlow::Node getAnExportedValue(Module mod, string name) {
```

Comment:
I like this idea.  
I'll look into it. 

I might end up opening up another PR and then rebasing this PR on top of that. 

> and then for backwards compatibility exports can be defined as this.getAnExportedValue().getAstNode()

There is complication. The `ast` in `exports(name, ast)` is often a property-write, and I would like the `getAnExportedValue()` predicate would return the RHS instead of the property-write itself.     
So defining `exports(name, ast)` as `ast = this.getAnExportedValue(name).getAstNode()` would not preserve behavior.

Reply:
Ok, regarding `exports` a less elegant alternative might be to inline the whole definition into its root definition and then deprecate it like that (as we can't override deprecated predicates).